src/fparser/two/utils.py

```diff
@@ -714,9 +714,15 @@ def match(
                     # We've found the enclosing end statement so break out
                     found_end = True
                     break
+
```

Comment:
Not strictly yours but since we're here pylint complains that the `elif` at L701 could be an `if`. Probably the same goes for L705.

Reply:
Modified.